File: spring-boot-api/src/main/java/com/vituary/hello/teams/Team.java
package com.vituary.hello.teams;

public class Team {
    private final String name;
    private final String arena;
    private final Address address;

    public Team(String name, String arena, Address address) {
        this.name = name;
        this.arena = arena;
        this.address = address;
    }

    public String getName() {
        return name;
    }

    public String getArena() {
        return arena;
    }

    public Address getAddress() {
        return address;
    }
}

File: react-ui/src/model/team.js
class TeamAddress {
  constructor(remote) {
    this.street = remote.street || null
    this.city = remote.city || null
    this.region = remote.region || null
    this.country = remote.country || null
    this.postalCode = remote.postalCode || null
  }
}

export default class Team {
  constructor(remote) {
    this.name = remote.name || null;
    this.arena = remote.arena || null;
    this.address = remote.address ? new TeamAddress(remote.address) : null
  }
}
